### PR TITLE
[GBM] Add meson / allow single plane mode

### PR DESCRIPTION
**Pull request by peak3d**
## Description
This PR enables support of the gbm system for odroid-c2 (and family)

- Add meson drm device
- Allow single plane (because AML drm has no overlay plane)

## Motivation and Context
Make AML devices ready for new linux kernel ( I use 4.15)

## How Has This Been Tested?
Odroid C2 / Linux Ubuntu Minimal Server / Kernel 4.15 / latest libMali with DRM support and DMA.

## Types of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: xbmc/windowing/gbm/DRMUtils.cpp [hunk 1]
```diff
@@ -374,53 +374,59 @@ bool CDRMUtils::GetPlanes()
 
   CLog::Log(LOGDEBUG, "CDRMUtils::%s - primary plane format: %c%c%c%c", __FUNCTION__, fourcc, fourcc >> 8, fourcc >> 16, fourcc >> 24);
 
-  // overlay plane
```
Comment (peak3d) [inline xbmc/windowing/gbm/DRMUtils.cpp]: Identation changed, looks worse as it is :-)

**Comment by lrusak** [comment]
generally I am ok with this but I don't like adding hacks to work around platforms
If you view this PR with whitespace off then it looks much better https://github.com/xbmc/xbmc/pull/13503/files?w=1

Do you know why the encoder id and crtc id is 0?

**Comment by lrusak** [comment]
Please don't merge this until we actually need it for something

**Comment by peak3d** [comment]
@lrusak I have placed the question at n.armstrong and I'm waiting for an answer.

**Comment by lrusak** [comment]
I think he is @superna9999 on github

**Comment by peak3d** [comment]
@lrusak connector id quirks reverted....

**Comment by lrusak** [comment]
What about just setting the overlay plane to the primary plane if an overlay plane cannot be found? This way you can remove the overlay plane checks.

**Comment by peak3d** [comment]
@lrusak ready to go?

**Comment by peak3d** [comment]
@lrusak overlay plane is now set to primary plane, any objections?

jenkins build this please

**Review by lrusak — APPROVED** [review]

**Comment by peak3d** [comment]
OSX64 test unrelated

**Pull request merged**